### PR TITLE
Expose temperature.fuentes in PVSystem and ModelChain

### LINKED ISSUE #1042: expose pvlib.temperature.fuentes in PVSystem and ModelChain
Follow up to #1032 and #1037

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] Closes #1042 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Basically just copied #897.  Happy to make changes if needed.  CI is probably going to complain about this until #1071 is fixed.

### CODE REVIEW

File: pvlib/pvsystem.py [hunk 1]
```diff
@@ -609,6 +609,33 @@ def faiman_celltemp(self, poa_global, temp_air, wind_speed=1.0):
...
+            Wind speed [m/s]
+
+        Returns
+        -------
+        temperature_cell : pandas Series
+            The modeled cell temperature [C]
+        """
+        kwargs = _build_kwargs([
+            'noct_installed', 'module_height', 'wind_height', 'emissivity',
+            'absorption', 'surface_tilt', 'module_width', 'module_length'],
```
Comment (cwhanse) [inline pvlib/pvsystem.py:634]: I think `'surface_tilt'` is a `PVSystem` attribute, not to be found in `temperature_model_parameters`

We might also consider getting `'module_width'` and `'module_length'` from `module_parameters`, although: the database doesn't supply these values and they aren't used anywhere else. `'module_height'` is really an attribute of the PV System.

For now, I'm OK expecting the `'module_'` values in `temperature_model_parameters` but they will move once another, non-temperature function needs this information - e.g., a bifacial irradiance function.

Comment (kanderso-nrel) [reply pvlib/pvsystem.py:634]: Valid points.  I unthinkingly pulled all the extra parameters from `temperature.fuentes` and dumped them there.  

I can imagine some people objecting to using Fuentes with the real `surface_tilt` value because they want the same behavior as the SSC implementation of PVWatts, which hardcodes the tilt at 30.  Would it make sense to use the `PVSystem` attributes by default but allow the user to override them by providing alternate values in the `temperature_model_parameters` dictionary?

Comment (cwhanse) [reply pvlib/pvsystem.py:634]: I don't have a better idea at the moment, than using  `temperature_model_parameters['surface_tilt']` as the signal to override. But I think it risks confusing users, to have two 'surface_tilt' quantities with different meanings. Maybe a Note in the `PVSystem` docstring and move ahead.

Comment (wholmgren) [reply pvlib/pvsystem.py:634]: Also something to consider when creating an Array class.

File: pvlib/pvsystem.py [hunk 2]
```diff
@@ -628,11 +628,24 @@ def fuentes_celltemp(self, poa_global, temp_air, wind_speed):
         -------
         temperature_cell : pandas Series
             The modeled cell temperature [C]
+
+        Notes
+        -----
+        The Fuentes thermal model uses the module surface tilt for convection
+        modeling. The SAM implementation of PVWatts hardcodes the value at 30
```
Comment (cwhanse) [inline pvlib/pvsystem.py:635]:
```suggestion
        modeling. The SAM implementation of PVWatts hardcodes the surface tilt value at 30
```

**Review by cwhanse — APPROVED** [review]

File: docs/sphinx/source/whatsnew/v0.8.1.rst [hunk 3]
```diff
@@ -13,6 +13,8 @@ Deprecations
 
 Enhancements
 ~~~~~~~~~~~~
+* Create :py:func:`~pvlib.pvsystem.PVSystem.fuentes_celltemp` and add `temperature_model='fuentes'``
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.8.1.rst:16]:
```suggestion
* Create :py:func:`~pvlib.pvsystem.PVSystem.fuentes_celltemp` and add ``temperature_model='fuentes'``
```

**Review by wholmgren — APPROVED** [review]
trivial doc fix @kanderso-nrel
